### ISSUE TITLE
--hipsycl-targets: inconvenient platform separator

### ISSUE THREAD

**Issue by al42and** [AdaptiveCpp/AdaptiveCpp#473]
Using semicolon as part of the `--hipsycl-targets` CLI argument is less than ideal:

- In bash (I guess, all POSIX shells), one needs to either quote the string or escape the semicolon.
- When using CMake, one has to both quote the string and escape the semicolon, e.g. `-DCMAKE_CXX_FLAGS='--hipsycl-targets=cuda:sm_75\;hip:gfx900'`.

These are, of course, not insurmountable, but require extra care when scripting the build and can cause certain frustration.

Maybe some other, safer separator would be better? E.g., `+` or `/`? 

Another solution would be to allow passing multiple `--hipsycl-target`, but that would require larger code changes and would lead to longer syclcc invocations, so probably not a good approach?

I can make the required patch if this concern is shared.

**Comment by illuhad** [comment]
I don't think putting in quotes is a big issue... A cmake invocation is not any more difficut. The correct way to invoke is to use e.g. `cmake -DHIPSYCL_TARGETS="omp;hip:gfx906"`.

Going through `CMAKE_CXX_FLAGS` is not the right way to use hipSYCL with cmake.

Given that any change to this breaks user-facing conventions, I'm not eager to make any changes here.

**Comment by al42and** [comment]
> I don't think putting in quotes is a big issue... 

Not a big issue, but can cause a bit more head-scratching than necessary when one starts with a single platform that works fine with `--hipsycl-targets=cuda:sm_75`, and then decides to add another one. Or when the build command is constructed through string manipulation in *sh.

In my experience, using `;` (and `&`) reliably causes issues for users who are not very proficient with shell.

> The correct way to invoke is to use e.g. cmake -DHIPSYCL_TARGETS="omp;hip:gfx906".

Thanks for explaining! I got the build working with mostly just passing `-DCMAKE_CXX_COMPILER=syclcc`, and did not notice that there is a proper CMake module to use.

> Given that any change to this breaks user-facing conventions, I'm not eager to make any changes here.

Several kinds of separators can be allowed, but I understand your reluctance to add more complications and variability to the interface!

Feel free to close this issue.

**Comment by illuhad** [comment]
> Several kinds of separators can be allowed, but I understand your reluctance to add more complications and variability to the interface!

Right. We could add synonyms, but I'm not sure which one might be a good fit. `+` and `-` are both potentially used to enable/disable compilation features in the hip toolchain as part of the target name. On the other hand, `:` also crept in.. e.g. `gfx908:xnack+`, so we might need a more general solution to specify target literals. Like brackets or quotes.
`/` seems unintuitive for a list of backends.
What might also work is that we improve the parsing long term such that `,` could also be used: `omp,hip:gfx900,gfx906,cuda:sm_60`.
